addons/io_hubs_addon/components/definitions/environment_settings.py

```diff
@@ -0,0 +1,74 @@
+from bpy.props import FloatProperty, EnumProperty, FloatVectorProperty, PointerProperty
+from bpy.types import Image
+
+from ...io.utils import gather_texture_property, gather_color_property
+from .hubs_component import HubsComponent
+from ..types import Category, PanelType, NodeType
+
+
+TOME_MAPPING = [("NoToneMapping", "None", "No tone mapping."),
+                ("LinearToneMapping", "Linear", "Linear tone mapping"),
+                ("ReinhardToneMapping", "ThreeJS 'Reinhard'",
+                 "ThreeJS 'Reinhard' tone mapping"),
+                ("CineonToneMapping", "ThreeJS 'Cineon'",
+                 "ThreeJS 'Cineon' tone mapping"),
+                ("ACESFilmicToneMapping", "ThreeJS 'ACES Filmic'",
+                 "ThreeJS 'ACES Filmic' tone mapping"),
+                ("LUTToneMapping", "Blender 'Filmic'", "Match Blender's Filmic tone mapping")]
+
+
+class EnvironmentSettings(HubsComponent):
+    _definition = {
+        'id': 'environment-settings',
+        'name': 'hubs_component_environment_settings',
+        'display_name': 'Environment Settings',
+        'category': Category.SCENE,
+        'node_type': NodeType.SCENE,
+        'panel_type': PanelType.SCENE,
+        'icon': 'WORLD'
+    }
+
+    toneMapping: EnumProperty(
+        name="Tone Mapping",
+        description="Tone Mapping",
+        items=TOME_MAPPING,
+        default="LUTToneMapping")
+
+    toneMappingExposure: FloatProperty(
+        name="Exposure", description="Exposure level of tone mapping", default=1.0, min=0.0, soft_min=0.0)
+
+    backgroundColor: FloatVectorProperty(name="Background Color",
+                                         description="Background Color",
+                                         subtype='COLOR',
+                                         default=(1.0, 1.0, 1.0, 1.0),
+                                         size=4,
+                                         min=0,
+                                         max=1)
+    backgroundTexture: PointerProperty(
+        name="Background Image",
+        description="An equirectangular image to use as the scene background",
+        type=Image
+    )
+
+    envMapTexture: PointerProperty(
+        name="EnvMap",
+        description="An equirectangular image to use as the default environment map for all objects",
+        type=Image
+    )
+
+    def gather(self, export_settings, object):
+        return {
+            'toneMapping': self.toneMapping,
+            'toneMappingExposure': self.toneMappingExposure,
+            'backgroundColor': gather_color_property(export_settings, object, self, 'backgroundColor'),
```

Comment:
That has to do with the fact that `backgroundTexture` and `envMapTexture` are defined as `Image` but need to be exported as textures. It happens the same in the reflection probe. Changing the type to `Texture`would require the user to create a texture from an image first before being able to select it.

Not sure what's the best path, thinking about it sounds like the later path is more correct although I think from the usability perspective probably loading and picking images seems more natural.

Reply:
Ah yeah. I forget why it was important that the component point at a texture in the gltf not an image, but there was some reason. I guess that makes sense. It might be that we always want to gather image properties as textures in which case we could just do that in the default infiltrator, but being more explicit like this for now makes sense.